mono/metadata/gc.c

```diff
@@ -943,11 +944,20 @@ mono_gc_init_finalizer_thread (void)
 	mono_error_assert_ok (error);
 }
 
+void
+reference_queue_mutex_init (void)
+{
+	if (mono_atomic_cas_i32 ((gint32 *)&reference_queue_mutex_inited, 0x1, 0x0) == 0x0) {
+		// If we are the initializer
```

Comment:
No need for `0x`.

Reply:
Why the cast? The type is already `gint32`.

Reply:
What happens to threads that don't get the cas, they just fall through and use it? Please use `pthread_once`, err, the mono wrapper of it (Win32 has an analog.

Reply:
If reference_queue_mutex is a pointer, the strongly consider check null, new local, cas it with global, free if race lost. This cuts out some subtle race conditions. Also barrier between init and cas maybe, maybe.